modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,9 +55,14 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
+    const descendants = SelectorFilter.descendants(toBlock, '*');
```

Comment:
This would break if the structure is complex say it shouldn't wrap the BR with both B and I in this case.
```
<p>
<b></b>
<i></i>
</p>
```

So we are likely only interested in the first or last branch of the tree also only inline elements. So I suggest doing a recursive function that traverses down on the first child and collects them and maybe filters out blocks.

So if the structure is `<p><b><i></i></b><span></span>` it would clone the B, I but not the SPAN.
Also if the structure is <div><p><b><i>...` then I think it should ignore the P element.

Reply:
How can this kind of markup appear, besides editing HTML manually? Just curious.

Updated the code with the traversing.

https://github.com/tinymce/tinymce/blob/5d854d3e550ce06fca195f6cac6b2f3ce42d4b40/modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts#L58-L64

Reply:
And added an additional inline element to the test as well to handle the case with several children.

Reply:
If you look at the implementation of `Empty.isEmpty`, Spocke's example should return true. How we might get there is another question :)
https://github.com/tinymce/tinymce/blob/ece3bf774b9ae20910d073af5351f162c4fb52d5/modules/tinymce/src/core/main/ts/dom/Empty.ts#L31-L71